discovery-provider/src/queries/queries.py

```diff
@@ -1872,3 +1872,146 @@ def get_top_followee_saves(type):
+                    User.user_id
+                ).order_by(
+                    desc(func.count(Follow.follower_user_id).label('follower_count')),
+                    User.user_id
+                )
+
+            # If the with_users flag is used, retrieve the user metadata
+            if with_users:
+                user_subquery = paginate_query(user_query).subquery('user_query')
+                user_query = session.query(User).filter(
+                    User.user_id.in_(user_subquery),
+                    User.is_current == True
+                )
+                users = user_query.all()
+                users = helpers.query_result_to_list(users)
+                user_ids = list(map(lambda user: user["user_id"], users))
+                users = populate_user_metadata(session, user_ids, users, None)
+                return api_helpers.success_response({ 'users': users })
+
+
+            # If the with_users flag is not set, respond with the user_ids
+            users = paginate_query(user_query).all()
+            user_ids = list(map(lambda user: user[0], users))
+
+            return api_helpers.success_response({ 'user_ids': user_ids })
+
+        # Associate the user w/ a genre by counting the total # of tracks per genre
+        # taking the genre w/ the most tracks (using genre name as secondary sort)  
+        user_genre_count_query = (
+            session.query(
+                User.user_id.label('user_id'),
+                Track.genre.label('genre'),
+                func.row_number().over(
+                        partition_by=User.user_id,
+                        order_by=(desc(func.count(Track.genre)), asc(Track.genre))
+                ).label("row_number")
+            ).join(
+                Track,
+                Track.owner_id == User.user_id
+            ).filter(
+                User.is_current == True,
+                User.is_creator == True,
+                Track.genre.isnot(None),
+                Track.is_unlisted == False,
+                Track.is_current == True,
+                Track.is_delete == False
+            ).group_by(
+                User.user_id,
+                Track.genre
+            ).order_by(
+                desc(func.count(Track.genre)), asc(Track.genre)
+            ).subquery('user_genre_count_query')
+        )
+
+        user_genre_query = (
+            session.query(
+                user_genre_count_query.c.user_id.label('user_id'),
+                user_genre_count_query.c.genre.label('genre'),
+            ).filter(
+                user_genre_count_query.c.row_number == 1
+            ).subquery('user_genre_query')
+        )
+
+        # Using the subquery of user to associated genre,
+        #   filter by the requested genres and 
+        #   sort by user follower count
+        user_genre_query = (
+            session.query(
+                user_genre_query.c.user_id.label('user_id')
+                # user_genre_query.c.genre.label('genre'),
```

Comment:
should these be commented out?

Reply:
Oh sorry, yeah this should be deleted. I was returning the userid & genre before (b/c it might be useful for a query w/ multiple genres), but not need.